File: templates/problems/18_bit_manipulation/01_single_number/go/solution.go
package single_number

// solveSingleNumber XORs all elements; duplicates cancel out. O(n) time, O(1) space.
func solveSingleNumber(nums []int) int {
	result := 0
	for _, n := range nums {
		result ^= n
	}
	return result
}

File: templates/problems/18_bit_manipulation/01_single_number/typescript/solution.ts
/**
 * XOR all elements; duplicates cancel out. O(n) time, O(1) space.
 */
export function solveSingleNumber(nums: number[]): number {
    let result = 0;
    for (const n of nums) {
        result ^= n;
    }
    return result;
}
